src/lasuite/malware_detection/backends/jcop.py

```diff
@@ -0,0 +1,222 @@
+            },
+            **kwargs,
+        )
+        raise MalwareDetectionInvalidAuthenticationError()
+
+    if response.status_code == HTTPStatus.OK:
+        content = response.json()
+        if not content["done"]:
+            # the analysis is not done yet, retry later
+            self.retry()
+            return
+
+        if content.get("error_code"):
+            status = ReportStatus.UNSAFE if content["is_malware"] else ReportStatus.UNKNOWN
+            callback(
+                file_path,
+                status,
+                error_info={
+                    "error": content["error"],
+                    "error_code": content["error_code"],
+                },
+                **kwargs,
+            )
+            return
+
+        if not content.get("is_malware"):
+            callback(file_path, ReportStatus.SAFE, error_info={}, **kwargs)
+            return
+
+    # Any other case, call the callback with an unknown error
+    callback(
+        file_path,
+        ReportStatus.UNKNOWN,
+        error_info={
+            "error": "Unknown treatment",
+            "error_code": response.status_code,
+        },
+        **kwargs,
+    )
+
+
+@shared_task(bind=True, max_retries=6, dont_autoretry_for=(MalwareDetectionInvalidAuthenticationError,))
+def trigger_new_analysis(self, file_path: str, **kwargs) -> None:
+    """Trigger a new analysis for a file."""
+    file = default_storage.open(file_path, "rb")
+
+    api_key = getattr(settings, setting_name("JCOP_API_KEY"))
+    base_url = getattr(settings, setting_name("JCOP_BASE_URL"))
+    callback = import_string(getattr(settings, setting_name("MALWARE_DETECTION_CALLBACK")))
+    timeout = getattr(settings, setting_name("JCOP_SUBMIT_TIMEOUT"), 10 * 60)
+
+    encoder = MultipartEncoder(
+        fields={
+            "file": (file.name, file),
+        }
+    )
+    try:
+        response = requests.post(
+            f"{base_url}/submit",
+            headers={
+                "X-Auth-Token": api_key,
+                "Accept": "application/json",
+                "Content-Type": encoder.content_type,
+            },
+            data=encoder,
+            timeout=(30, timeout),
+        )
+    except requests.exceptions.RequestException as exc:
+        logger.error("Error submitting file to JCOP: %s", exc)
+        if self.request.retries >= self.max_retries:
+            callback(
+                file_path,
+                ReportStatus.UNKNOWN,
+                error_info={
+                    "error": "Max retries triggering new analysis exceeded",
+                    "error_code": 5000,
+                },
+                **kwargs,
+            )
+        raise self.retry(exc=exc) from exc
+    finally:
+        file.close()
+
+    if response.status_code == HTTPStatus.OK:
+        content = response.json()
+        analyse_file_async.apply_async(countdown=30, args=(file_path,), kwargs={"file_hash": content["id"], **kwargs})
```

Comment:
I'm not familiar with the difference between `delay` and `apply_async`, should we be consistent in this module?

Reply:
apply_async allow to configure the delay to apply before running the task. `delay` is a shortcut of `apply_async`.
Here I can't use `delay`